### PR TITLE
chore: Migrate users to GitHub identity only

### PR DESCRIPTION
**Pull request by tumido**
Part 2 of 2 of https://github.com/operate-first/apps/issues/791

Take encrypted groups enriched by https://github.com/operate-first/apps/pull/841 and remove emails. Then decrypt and store them in base instead of encrypted overlays.

/hold

**Don't merge until SSO transition is complete**
**Please leave this on hold until we decide to disable MOC SSO.**

For removal of obsolete team members please comment on https://github.com/operate-first/support/issues/357

### CODE REVIEW

File: cluster-scope/base/user.openshift.io/groups/data-science/group.yaml [hunk 1]
```diff
@@ -2,5 +2,16 @@
...
+    - oindrillac
+    - aakankshaduggal
+    - Shreyanand
+    - MichaelClifford
+    - chauhankaranraj
+    - SankBad
+    - goern
+    - pacospace
+    - durandom
+    - martinpovolny
```
Comment (4n4nd) [inline cluster-scope/base/user.openshift.io/groups/data-science/group.yaml:16]: I think we can remove `SankBad` and `martinpovolny`

File: cluster-scope/base/user.openshift.io/groups/operate-first/group.yaml [hunk 2]
```diff
@@ -2,5 +2,19 @@
...
+    - 4n4nd
+    - HumairAK
+    - hemajv
+    - durandom
+    - martinpovolny
+    - tumido
+    - goern
+    - harshad16
+    - pacospace
+    - ipolonsk
```
Comment (4n4nd) [inline cluster-scope/base/user.openshift.io/groups/operate-first/group.yaml:18]: we can remove `ipolonsk`

File: cluster-scope/base/user.openshift.io/groups/prometheus-anomaly-detector/group.yaml [hunk 3]
```diff
@@ -2,4 +2,7 @@ apiVersion: user.openshift.io/v1
 kind: Group
 metadata:
     name: prometheus-anomaly-detector
-users: []
+users:
+    # FIXME 1 account missing
```
Comment (4n4nd) [inline cluster-scope/base/user.openshift.io/groups/prometheus-anomaly-detector/group.yaml:6]: these two users should be fine

**Review by 4n4nd — CHANGES_REQUESTED** [review]

File: cluster-scope/base/user.openshift.io/groups/data-science/group.yaml (hunk 1 above)
Comment (tumido) [reply cluster-scope/base/user.openshift.io/groups/data-science/group.yaml:16]: @4n4nd hm... what about we do a 1to1 mapping first, so we can be really sure we did not forget anyone and do a clean up later?

I'm worried we might accidentally remove somebody.

File: cluster-scope/base/user.openshift.io/groups/operate-first/group.yaml (hunk 2 above)
Comment (tumido) [reply cluster-scope/base/user.openshift.io/groups/operate-first/group.yaml:18]: same as above, what about we do that in a follow up PR?

File: cluster-scope/base/user.openshift.io/groups/prometheus-anomaly-detector/group.yaml (hunk 3 above)
Comment (tumido) [reply cluster-scope/base/user.openshift.io/groups/prometheus-anomaly-detector/group.yaml:6]: But can you get the github username for the 3rd person anyways :smile: ?

**Comment by tumido** [comment]
Rebased on #841

File: cluster-scope/base/user.openshift.io/groups/open-aiops/group.yaml [hunk 4]
```diff
@@ -1,6 +1,10 @@
----
 apiVersion: user.openshift.io/v1
 kind: Group
 metadata:
   name: open-aiops
-users: []
+users:
+  - davidohana
+  - ronensc
+  # FIXME: account missing
```
Comment (4n4nd) [inline cluster-scope/base/user.openshift.io/groups/open-aiops/group.yaml:8]: can be removed, was confirmed by a team member

**Review by 4n4nd — APPROVED** [review]

**Comment by tumido** [comment]
Please leave this on hold until we decide to disable MOC SSO.

File: cluster-scope/base/user.openshift.io/groups/operate-first/group.yaml [hunk 5]
```diff
@@ -1,6 +1,19 @@
...
   name: operate-first
-users: []
+users:
+  - anishasthana
+  - accorvin
+  - 4n4nd
+  - humairak
+  - hemajv
+  - durandom
+  - martinpovolny
```
Comment (anishasthana) [inline cluster-scope/base/user.openshift.io/groups/operate-first/group.yaml:12]: martin shouldn't be on this group now right?

Comment (tumido) [reply cluster-scope/base/user.openshift.io/groups/operate-first/group.yaml:12]: We're not doing a cleanup yet. Right now we should aim for 1 to 1 mapping of current RBAC. Once we have every group in plaintext we can start with a cleanup. :slightly_smiling_face: 

I'll create additional issue for RBAC revisiting and aggregate your comment in there, sounds good? :+1:

Comment (tumido) [reply cluster-scope/base/user.openshift.io/groups/operate-first/group.yaml:12]: https://github.com/operate-first/support/issues/357

Comment (anishasthana) [reply cluster-scope/base/user.openshift.io/groups/operate-first/group.yaml:12]: Sounds good! Thanks

File: cluster-scope/base/user.openshift.io/groups/open-aiops/group.yaml (hunk 4 above)
Comment (tumido) [reply cluster-scope/base/user.openshift.io/groups/open-aiops/group.yaml]: updated

**Comment by tumido** [comment]
/unhold

I think this is ready to be reviewed now

**Review by 4n4nd — APPROVED** [review]
/lgtm

**Pull request merged**